packml_sm/test/state_machine.cpp

```diff
@@ -227,4 +229,81 @@ TEST(Packml_CC, state_diagram)
   EXPECT_FALSE(sm->isActive());
   ROS_INFO_STREAM("State diagram test complete");
 }
+
+TEST(Packml_CC, load_stats)
+{
+  ROS_INFO_STREAM("CONTINUOUS CYCLE::Load stats");
+  std::shared_ptr<AbstractStateMachine> sm = PackmlStateMachineContinuous::spawn();
+  EXPECT_FALSE(sm->isActive());
+  sm->setExecute(std::bind(success));
+  sm->activate();
+  ros::Duration(1.0).sleep();  // give time to start
```

Comment:
Did you do any analysis on the duration here? Would it be better to loop on sm->isActive until some timeout instead? Hardcoded durations, especially nice round ones like this, are a minor red flag.

Reply:
Yeah, tbh I copied that from the existing unit tests. I do think a loop makes more sense, but after thinking about it more the load stats functionality should be totally agnostic of the current state, so I removed all the state changes and checks from the unit test, making it a pure test of loadStats.